proxy/healthy_endpoints_test.go

```diff
@@ -150,142 +161,84 @@ func TestPHCForSingleHealthyEndpoint(t *testing.T) {
-		})
-	})
-
-	t.Run("consistent hash with balance factor", func(t *testing.T) {
-		mockMetrics, ps := setupProxy(t, fmt.Sprintf(`* -> backendTimeout("5ms") -> consistentHashKey("${request.header.ConsistentHashKey}") -> consistentHashBalanceFactor(1.25) -> <consistentHash, "%s", "%s", "%s">`,
-			services[0], services[1], services[2]))
-		failedReqs := sendGetRequests(t, ps)
-		assert.InDelta(t, 0, failedReqs, 0.1*float64(nRequests))
-		mockMetrics.WithCounters(func(counters map[string]int64) {
-			assert.InDelta(t, float64(nRequests), float64(counters["passive-health-check.endpoints.dropped"]), 0.3*float64(nRequests)) // allow 30% error
-		})
-	})
-}
-
-func TestPHCForMultipleHealthyAndMultipleUnhealthyEndpoints(t *testing.T) {
-	services := setupServices(t, 2, 2)
-	for _, algorithm := range []string{"random", "roundRobin", "powerOfRandomNChoices"} {
-		t.Run(algorithm, func(t *testing.T) {
-			mockMetrics, ps := setupProxy(t, fmt.Sprintf(`* -> backendTimeout("5ms") -> consistentHashKey("${request.header.ConsistentHashKey}") -> <%s, "%s", "%s", "%s", "%s">`,
-				algorithm, services[0], services[1], services[2], services[3]))
-
-			va := fireVegeta(t, ps, 3000, 1*time.Second, 5*time.Second)
-
-			count200, ok := va.CountStatus(200)
-			assert.True(t, ok)
-
-			count504, ok := va.CountStatus(504)
-			assert.True(t, ok)
-
-			failedRequests := math.Abs(float64(va.TotalRequests())) - float64(count200)
-			t.Logf("total requests: %d, count200: %d, count504: %d, failedRequests: %f", va.TotalRequests(), count200, count504, failedRequests)
-
-			assert.InDelta(t, float64(count504), failedRequests, 5)
-			assert.InDelta(t, 0, float64(failedRequests), 0.3*float64(va.TotalRequests()))
-			mockMetrics.WithCounters(func(counters map[string]int64) {
-				assert.InDelta(t, 2*float64(va.TotalRequests()), float64(counters["passive-health-check.endpoints.dropped"]), 0.6*float64(va.TotalRequests()))
+			t.Run("consistent hash with balance factor", func(t *testing.T) {
+				endpointRegistery := routing.NewEndpointRegistry(routing.RegistryOptions{
+					PassiveHealthCheckEnabled:     true,
+					StatsResetPeriod:              1 * time.Second,
+					MinRequests:                   1, // with 2 test case fails on github actions
+					MaxHealthCheckDropProbability: 0.95,
+					MinHealthCheckDropProbability: 0.01,
+				})
+				mockMetrics, ps := setupProxyWithCustomEndpointRegisty(t, fmt.Sprintf(`* -> backendTimeout("5ms") -> consistentHashKey("${request.header.ConsistentHashKey}") -> consistentHashBalanceFactor(1.25) -> <consistentHash, %s>`,
+					servicesString), endpointRegistery)
+				failedReqs := sendGetRequests(t, ps)
+				assert.InDelta(t, 0, failedReqs, 0.2*float64(nRequests))
+				mockMetrics.WithCounters(func(counters map[string]int64) {
+					assert.InDelta(t, float64(tt.unhealthy*nRequests), float64(counters["passive-health-check.endpoints.dropped"]), 0.3*float64(nRequests)*float64(tt.unhealthy))
```

Comment:
minor: I see `0.3*float64(nRequests)*float64(tt.unhealthy)` here and `0.3*float64(tt.unhealthy*nRequests)` above. Both options are fine for me, but I think they should be the same for all test cases to increase readability.